app/services/organizations/default_pet_task_service.rb

```diff
@@ -0,0 +1,28 @@
+# Service object that creates the default pet tasks for a newly created Pet
+class Organizations::DefaultPetTaskService
+  def initialize(pet)
+    @pet = pet
+    @default_pet_tasks = DefaultPetTask.all
+  end
+
+  def default_tasks_array
+    @default_pet_tasks.map do |task|
+      {
+        pet_id: @pet.id,
+        name: task.name,
+        description: task.description,
+        completed: false
+      }
+    end
+  end
+
+  def create_tasks
+    ActiveRecord::Base.transaction do
+      Task.create!(default_tasks_array)
+    end
```

Comment:
Mentioned above you'll want to let the raised error pass through to the top so it triggers the Rollback

Reply:
got it so I have remove the rescue block and now have:

```
 def create_tasks
    ActiveRecord::Base.transaction do
      Task.create!(default_tasks_array)
    end
  end
```